libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -1794,9 +1836,19 @@ CK_DECLARE_FUNCTION( CK_RV, C_OpenSession )( CK_SLOT_ID slotID,
 CK_DECLARE_FUNCTION( CK_RV, C_CloseSession )( CK_SESSION_HANDLE hSession )
 {
     P11Session_t * pxSession = prvSessionPointerFromHandle( hSession );
-    CK_RV xResult = prvCheckValidSessionAndModule( pxSession );
+    CK_RV xResult = CKR_OK;
```

Comment:
Does removing this check change the behavior?

Reply:
it's slightly different as I added a check to see if the session was opened or not, this wasn't necessary when it was added dynamically, as if the memory was valid we knew it was opened. The function has almost the same logic, except it checks to see if the session was opened before clearing the memory